subgraphs/metrom/src/mappings/v2.ts

```diff
@@ -0,0 +1,168 @@
+import { Address, BigInt, Bytes, ethereum } from "@graphprotocol/graph-ts";
+import {
+    CreateRewardsCampaign,
+    CreatePointsCampaign,
+    SetMinimumFeeTokenRate,
+} from "../../generated/MetromV2/MetromV2";
+import {
+    RewardsCampaign,
+    PointsCampaign,
+    CreateRewardsCampaignEvent,
+    CreatePointsCampaignEvent,
+    Reward,
+    CreatedCampaignReward,
+    SetMinimumFeeTokenRateEvent,
+} from "../../generated/schema";
+import { METROM_ADDRESS } from "../addresses";
+import {
+    getEventId,
+    getMetromOrThrow,
+    getOrCreateClaimableFee,
+    getOrCreateToken,
+    getOrCreateTransaction,
+    getRewardId,
+    getOrCreateWhitelistedFeeToken,
+    BI_1,
+    BI_0,
+} from "../commons";
+
+export function handleCreateRewardsCampaign(
+    event: CreateRewardsCampaign,
+): void {
+    let metrom = getMetromOrThrow();
+    metrom.campaignsAmount = metrom.campaignsAmount.plus(BI_1);
+    metrom.save();
+
+    let transaction = getOrCreateTransaction(event);
+
+    let campaign = new RewardsCampaign(event.params.id);
+    campaign.transaction = transaction.id;
+    campaign.metrom = metrom.id;
+    campaign.creationBlockNumber = event.block.number;
+    campaign.creationTimestamp = event.block.timestamp;
+    campaign.owner = event.params.owner;
+    campaign.pendingOwner = Address.zero();
+    campaign.from = event.params.from;
+    campaign.to = event.params.to;
+    campaign.kind = BI_1;
```

Comment:
Shouldn't we get the `kind` from the params?

Reply:
This is the Metrom v2 mapper, and in the previous version of the contract we did not have a concept of campaign that rewarded anything else than AMM liquidity. The AMM liquidity campaign kind is 1, so in this case we hardcode it. In the v3 mapper, instead, we take the `kind` attribute's value from the event's params.